pkg/space-manager/provider.go

```diff
@@ -65,6 +67,13 @@ func newProviderClient(providerDesc *spacev1alpha1apis.SpaceProviderDesc) spacep
 		pClient = kindprovider.New(providerDesc.Name)
 	case spacev1alpha1apis.KubeflexProviderType:
 		pClient = kflexprovider.New(providerDesc.Name)
+	case spacev1alpha1apis.KcpProviderType:
+		pClient, err := providerkcp.New(providerDesc.Spec.Config)
```

Comment:
this is strange that only one returns error. Let's change all providers to return err and move the error handling outside the case?

Reply:
But other providers doesn't have any error case. New(...) is not part of an interface. The parameters are different and the return values are different.

Reply:
Let's add an issue to try and unify this. It may be part of the REST change